src/mcts/search.h

```diff
@@ -188,6 +191,9 @@ class Search {
 
   std::atomic<int> pending_searchers_{0};
 
+  std::vector<std::pair<Node*, int>> shared_collisions_
```

Comment:
I'm expecting it would be slower.

Reply:
Another option would be to have `tiny_map` which exposes map-like interface but uses vector inside. It already came up several times so one day it will be implemented I guess. It's not clear whether it would be any noticeable speedup in this case though.

Reply:
I'm not sure that the vector inside would be reliably small enough - especially for people with 4x gpus.